File: app/src/main/java/com/hex/evegate/api/dto/Song.java
package com.hex.evegate.api.dto;

import com.google.gson.annotations.SerializedName;

public class Song {
    @SerializedName("art") private String art;

    @SerializedName("artist") private String artist;

    @SerializedName("album") private String album;

    @SerializedName("custom_fields") private String[] custom_fields;

    @SerializedName("id") private String id;

    @SerializedName("text") private String text;

    @SerializedName("title") private String title;

    @SerializedName("lyrics") private String lyrics;

    public String getArt ()
    {
        return art;
    }

    public void setArt (String art)
    {
        this.art = art;
    }

    public String getArtist ()
    {
        return artist;
    }

    public void setArtist (String artist)
    {
        this.artist = artist;
    }

    public String getAlbum ()
    {
        return album;
    }

    public void setAlbum (String album)
    {
        this.album = album;
    }

    public String[] getCustom_fields ()
    {
        return custom_fields;
    }

    public void setCustom_fields (String[] custom_fields)
    {
        this.custom_fields = custom_fields;
    }

    public String getId ()
    {
        return id;
    }

    public void setId (String id)
    {
        this.id = id;
    }

    public String getText ()
    {
        return text;
    }

    public void setText (String text)
    {
        this.text = text;
    }

    public String getTitle ()
    {
        return title;
    }

    public void setTitle (String title)
    {
        this.title = title;
    }

    public String getLyrics ()
    {
        return lyrics;
    }

    public void setLyrics (String lyrics)
    {
        this.lyrics = lyrics;
    }

    @Override
    public String toString()
    {
        return "ClassPojo [art = "+art+", artist = "+artist+", album = "+album+", custom_fields = "+custom_fields+", id = "+id+", text = "+text+", title = "+title+", lyrics = "+lyrics+"]";
    }
}

File: app/src/main/java/com/hex/evegate/ui/mvp/model/Song.kt
package com.hex.evegate.ui.mvp.model

import com.google.gson.annotations.SerializedName

data class Song (
    @SerializedName("art") var art: String,
    @SerializedName("artist") var artist: String,
    @SerializedName("album") var album: String,
    @SerializedName("custom_fields") var custom_fields: Array<String>,
    @SerializedName("id") var id: String,
    @SerializedName("text") var text: String,
    @SerializedName("title") var title: String,
    @SerializedName("lyrics") var lyrics: String
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Song

        if (art != other.art) return false
        if (artist != other.artist) return false
        if (album != other.album) return false
        if (!custom_fields.contentEquals(other.custom_fields)) return false
        if (id != other.id) return false
        if (text != other.text) return false
        if (title != other.title) return false
        if (lyrics != other.lyrics) return false

        return true
    }

    override fun hashCode(): Int {
        var result = art.hashCode()
        result = 31 * result + artist.hashCode()
        result = 31 * result + album.hashCode()
        result = 31 * result + custom_fields.contentHashCode()
        result = 31 * result + id.hashCode()
        result = 31 * result + text.hashCode()
        result = 31 * result + title.hashCode()
        result = 31 * result + lyrics.hashCode()
        return result
    }
}
